File: store-app/src/main/java/com/comp5348/store/service/OrderService.java
package com.comp5348.store.service;

import com.comp5348.store.dto.CreateOrderRequest;
import com.comp5348.store.dto.OrderDTO;
import com.comp5348.store.entity.OrderStatus;

import java.util.List;

public interface OrderService {
    
    /**
     * Create an order
     */
    OrderDTO createOrder(CreateOrderRequest request);
    
    /**
     * Get an order by ID
     */
    OrderDTO getOrderById(Long orderId);
    
    /**
     * Get all orders for a user
     */
    List<OrderDTO> getOrdersByUserId(Long userId);
    
    /**
     * Get all orders
     */
    List<OrderDTO> getAllOrders();
    
    /**
     * Get orders by status
     */
    List<OrderDTO> getOrdersByStatus(OrderStatus status);
    
    /**
     * Update order status
     */
    OrderDTO updateOrderStatus(Long orderId, OrderStatus newStatus);
    
    /**
     * Cancel an order
     */
    OrderDTO cancelOrder(Long orderId);
}

File: store-frontend/src/services/orderService.ts
import api from './api';
import { Order, CreateOrderRequest, OrderStatus } from '../types';

export const orderService = {
  createOrder: async (orderData: CreateOrderRequest): Promise<Order> => {
    const response = await api.post('/orders', orderData);
    return response.data;
  },

  getOrderById: async (orderId: number): Promise<Order> => {
    const response = await api.get(`/orders/${orderId}`);
    return response.data;
  },

  getOrdersByUserId: async (userId: number): Promise<Order[]> => {
    const response = await api.get(`/orders/user/${userId}`);
    return response.data;
  },

  getAllOrders: async (): Promise<Order[]> => {
    const response = await api.get('/orders');
    return response.data;
  },

  getOrdersByStatus: async (status: OrderStatus): Promise<Order[]> => {
    const response = await api.get(`/orders/status/${status}`);
    return response.data;
  },

  updateOrderStatus: async (orderId: number, status: OrderStatus): Promise<Order> => {
    const response = await api.put(`/orders/${orderId}/status?status=${status}`);
    return response.data;
  },

  cancelOrder: async (orderId: number): Promise<Order> => {
    const response = await api.put(`/orders/${orderId}/cancel`);
    return response.data;
  },
};

